src/config.rs

```diff
@@ -69,6 +189,69 @@ impl Options for DefaultOptions {
     }
 }
 
+/// A SizeLimit that restricts serialized or deserialized messages from
+/// exceeding a certain byte length.
+#[derive(Copy, Clone)]
+pub struct Bounded(pub u64);
+
+/// A SizeLimit without a limit!
+/// Use this if you don't care about the size of encoded or decoded messages.
+#[derive(Copy, Clone)]
+pub struct Infinite;
+
+impl SizeLimit for Bounded {
+    #[inline(always)]
+    fn add(&mut self, n: u64) -> Result<()> {
+        if self.0 >= n {
+            self.0 -= n;
+            Ok(())
+        } else {
+            Err(Box::new(ErrorKind::SizeLimit))
+        }
+    }
+
+    #[inline(always)]
+    fn limit(&self) -> Option<u64> {
+        Some(self.0)
+    }
+}
+
+impl SizeLimit for Infinite {
+    #[inline(always)]
+    fn add(&mut self, _: u64) -> Result<()> {
+        Ok(())
+    }
+
+    #[inline(always)]
+    fn limit(&self) -> Option<u64> {
+        None
+    }
+}
+
+/// Little-endian byte ordering.
+#[derive(Copy, Clone)]
+pub struct LittleEndian;
```

Comment:
What is the reason for introducing new types for these endians? It might be better to just re-export the `byteorder` ones and implement the `BincodeByteOrder` trait on those.

Reply:
Implementing `impl<T> BincodeByteOrder for T where T: byteorder::ByteOrder { type Endian = T; }` is probably sufficient

Reply:
I created my own types in case, for some reason, I would need to add data to them in the future. Of course thinking about it, I wouldn't be able to anyways.

The custom BincodeByteOrder trait is necessary though. Since I probably will need to add methods to it at a future date.